Jenkinsfile

```diff
@@ -0,0 +1,11 @@
+#!groovy
+node {
+  stage 'checkout source and unit test'
+  checkout scm
+  sh 'git rev-parse --short HEAD > commit'
+  def sha1 = readFile('commit').trim()
```

Comment:
Is Jenkins single-threaded, or is there a race condition here where two different builds could start close together, with one over-writing the "commit" file between the time the other writes it and it reads it?